src/backend/BackendFactory.h

```diff
@@ -4,30 +4,24 @@
 #include <boost/algorithm/string.hpp>
 #include <backend/BackendInterface.h>
 #include <backend/CassandraBackend.h>
+#include <config/Config.h>
 
 namespace Backend {
 std::shared_ptr<BackendInterface>
-make_Backend(boost::asio::io_context& ioc, boost::json::object const& config)
+make_Backend(boost::asio::io_context& ioc, clio::Config const& config)
 {
     BOOST_LOG_TRIVIAL(info) << __func__ << ": Constructing BackendInterface";
 
-    boost::json::object dbConfig = config.at("database").as_object();
-
-    bool readOnly = false;
-    if (config.contains("read_only"))
-        readOnly = config.at("read_only").as_bool();
-
-    auto type = dbConfig.at("type").as_string();
-
+    auto readOnly = config.value_or("read_only", false);
+    auto type = config.value<std::string>("database.type");
     std::shared_ptr<BackendInterface> backend = nullptr;
 
     if (boost::iequals(type, "cassandra"))
     {
+        auto cfg = config.section("database." + type);
         if (config.contains("online_delete"))
-            dbConfig.at(type).as_object()["ttl"] =
-                config.at("online_delete").as_int64() * 4;
-        backend = std::make_shared<CassandraBackend>(
-            ioc, dbConfig.at(type).as_object());
+            cfg.set_value("ttl", config.value<int64_t>("online_delete") * 4);
```

Comment:
So this online_delete functionality is not actually currently supported. The whole thing is a WIP and needs more testing. It seems this is the only reason we need `set_value`. I'd rather refactor this and remove `set_value`. We can pass the `ttl` in as a constructor param, or we can just require `ttl` be specified in the cassandra config.

Reply:
great. i will happily remove 'set_value' and this param